app/__init__.py

```diff
@@ -12,23 +12,33 @@
 
 def create_app(test_config=None):
     app = Flask(__name__)
-    app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
 
-    if test_config is None:
-        app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
+    if not test_config:
+        app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+        app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get(
             "SQLALCHEMY_DATABASE_URI")
+            
     else:
         app.config["TESTING"] = True
+        app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
         app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
             "SQLALCHEMY_TEST_DATABASE_URI")
+    app.config['JSON_SORT_KEYS']=False 
 
     # Import models here for Alembic setup
     from app.models.task import Task
     from app.models.goal import Goal
 
+
+
     db.init_app(app)
     migrate.init_app(app, db)
 
     # Register Blueprints here
+    from .routes import tasks_bp
+    app.register_blueprint(tasks_bp)
+
+    from .routes import goals_bp
+    app.register_blueprint(goals_bp)
```

Comment:
You can reduce redundancy here by combining imports like so:

```suggestion
    from .routes import tasks_bp, goals_bp
    app.register_blueprint(tasks_bp)
    app.register_blueprint(goals_bp)
```